UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneExporter.cs

```diff
@@ -96,21 +125,175 @@ public void SaveGLTFandBin(string path, string fileName)
 			gltfFile.Close();
 			binFile.Close();
 #endif
-
-			foreach (var image in _images)
-			{
-				Debug.Log(image.name);
-				var renderTexture = RenderTexture.GetTemporary(image.width, image.height);
-				Graphics.Blit(image, renderTexture);
-				RenderTexture.active = renderTexture;
-				var exportTexture = new Texture2D(image.width, image.height);
-				exportTexture.ReadPixels(new Rect(0, 0, renderTexture.width, renderTexture.height), 0, 0);
-				exportTexture.Apply();
-				File.WriteAllBytes(Path.Combine(path, image.name + ".png"), exportTexture.EncodeToPNG());
-			}
+            exportImages(path);
+			
 		}
 
-		private SceneId ExportScene(string name, Transform[] rootObjTransforms)
+        private void exportImages(string outputPath)
+        {
+            for (int t = 0; t < _imageInfos.Count; ++t)
+            {
+                var image = _imageInfos[t].texture;
+                int height = image.height;
+                int width = image.width;
+
+                switch (_imageInfos[t].textureMapType)
+                {
+                    case TextureMapType.MetallicGloss:
+                        exportMetallicGlossTexture(image, outputPath);
+                        break;
+                    default:
+                        exportTexture(image, outputPath);
+                        break;
+                }
+            }
+        }
+
+        private void exportMetallicGlossTexture(Texture2D texture, string outputPath)
+        {
+            var textureWidth = texture.width;
+            var textureHeight = texture.height;
+            Color[] outputColors = new Color[textureWidth * textureHeight];
```

Comment:
Just make outputColors an out parameter to the addtexturepixels function and have it create the array. The preinitialization with default values shouldn't be needed if you are filling it all in the calling function

Reply:
These are removed, now I am using Graphics.blit.